src/System.Net.Sockets/tests/FunctionalTests/AgnosticListenerTest.cs

```diff
@@ -33,59 +33,56 @@ public void Create_Success()
 
         [OuterLoop] // TODO: Issue #11345
         [Fact]
-        public void ConnectWithV4_Success()
+        public async Task ConnectWithV4_Success()
         {
-            int port;
-            TcpListener listener = SocketTestExtensions.CreateAndStartTcpListenerOnAnonymousPort(out port);
-            IAsyncResult asyncResult = listener.BeginAcceptTcpClient(null, null);
+            TcpListener listener = SocketTestExtensions.CreateAndStartTcpListenerOnAnonymousPort(out int port);
+            Task<TcpClient> acceptTask = Task.Factory.FromAsync(listener.BeginAcceptTcpClient(null, null), listener.EndAcceptTcpClient);
 
             TcpClient client = new TcpClient(AddressFamily.InterNetwork);
-            client.ConnectAsync(IPAddress.Loopback, port).GetAwaiter().GetResult();
+            await client.ConnectAsync(IPAddress.Loopback, port);
 
-            TcpClient acceptedClient = listener.EndAcceptTcpClient(asyncResult);
+            TcpClient acceptedClient = await acceptTask;
```

Comment:
Nit: It'd be better to use the WhenAllOrAnyFailed helper here with the ConnectAsync and acceptTasks.